### PR TITLE
Sector Nord AG: Fixed Calendar appointment plugin AppointmentID data type.

### PR DESCRIPTION
**Pull request by LuBroering**
## Proposed change
Change of the data type for the appointment_id in the table calendar_appointment_plugin. 

## Type of change
- '1 - 💎 code quality'      - Code quality improvements to existing code or addition of unit tests

## Additional information
We had an issue with the small int on productive systems. 

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [ ] Local ZnunyCodePolicy passed.(❕)
- [ ] Local UnitTests / Selenium passed.(❕)
- [ ] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by jepf** [comment]
@LuBroering 

Thank you for the pull request and sorry for the delay.
May I kindly ask you to resolve the merge conflict that popped up in the meantime? I will merge it then.

Please be aware that dev branch now is the upcoming 6.5.1 and currently does not contain any db upgrade migrations. 
So you would have to also add the module to the task list in scripts/Migration/Znuny/UpgradeDatabaseStructure.pm.

Thank you.

**Review by jepf — APPROVED** [review]

**Comment by dennykorsukewitz** [comment]
Hi

This branch has conflicts that must be resolved.

Thank you! :rocket:

**Comment by dennykorsukewitz** [comment]
Hi @LuBroering 

This branch has conflicts that must be resolved.

Thank you! 🚀

**Comment by LuBroering** [comment]
Hi @dennykorsukewitz,

the conflicts are resolved.

File: scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# --
+# Copyright (C) 2021-2022 Znuny GmbH, https://znuny.org/
```
Comment (dennykorsukewitz) [inline scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm]:
```suggestion
# Copyright (C) 2021 Znuny GmbH, https://znuny.org/
```

File: scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+use strict;
+use warnings;
+
+use parent qw(scripts::Migration::Base);
+
+our @ObjectDependencies = (
+    'Kernel::System::DB',
+);
+=head1 SYNOPSIS
```
Comment (dennykorsukewitz) [inline scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm]:
```suggestion
);

=head1 SYNOPSIS
```

File: scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+sub _IncreaseColumnsSize {
+    my ( $Self, %Param ) = @_;
+
+    my @XMLStrings = (
+        '<TableAlter Name="calendar_appointment_plugin">
+            <ColumnChange NameOld="appointment_id" NameNew="appointment_id" Required="true" Type="BIGINT"/>
+        </TableAlter>'
+        
+    );
```
Comment (dennykorsukewitz) [inline scripts/Migration/Znuny/UpgradeDatabaseStructure/CalendarAppointmentID.pm]:
```suggestion
        </TableAlter>'
    );
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]

**Review by dennykorsukewitz — APPROVED** [review]

**Pull request merged**